app/configurator/components/add-dataset-dialog.tsx

```diff
@@ -483,6 +485,15 @@ export const PreviewDataTable = ({
     });
   };
 
+  const previewObservations = useMemo(() => {
+    const data = observations.data?.dataCubesObservations.data ?? [];
+    const bestObservation = maxBy(data, (obs) => {
+      return allColumns.reduce((acc, dim) => acc + (obs[dim.iri] ? 1 : 0), 0);
+    });
+    const index = bestObservation ? data.indexOf(bestObservation) : 0;
+    return data.slice(index, index + 8);
+  }, [allColumns, observations.data?.dataCubesObservations.data]);
```

Comment:
👌

Reply:
That's probably not going to happen, but if the best observation is the last one, would we only see one row?

Reply:
Thanks, I handle the edge case now 🙇 https://github.com/visualize-admin/visualization-tool/pull/1519/commits/4cfbeb87822a60eda69d9973a468dc5cc42a2a02